File: 2442-number-of-arithmetic-triplets/number-of-arithmetic-triplets.kt
class Solution {
    fun arithmeticTriplets(nums: IntArray, diff: Int): Int {
        val set :HashSet<Int> = nums.toHashSet()
        var count = 0
        for (i in nums.indices) {
            if (set.contains(nums[i] - diff) && set.contains(nums[i] - 2 * diff)) {
                count++
            }
        }
        return count
    }
}

File: 2442-number-of-arithmetic-triplets/number-of-arithmetic-triplets.swift
class Solution {
    func arithmeticTriplets(_ nums: [Int], _ diff: Int) -> Int {
        var answer = 0

        nums.forEach {
            if nums.contains($0 + diff), nums.contains($0 + diff + diff) {
                answer += 1
            }
        }

        return answer
    }
}